### ISSUE TITLE
[Feature Request] Stop tuning upon optimization convergence

### ISSUE THREAD

**Issue by rohan-gt** [ray-project/tune-sklearn#98]
Is it possible enable early stopping to any algorithm that does not have partial_fit (eg. LogisticRegression or RandomForest) just by looking at the train and test (CV score) score progression across the trials?

**Comment by richardliaw** [comment]
@rohan-gt good question! Can you clarify what you mean by "early stopping"? Do you mean:
1. Stop the hyperparameter sweep early, or
2. Stop the training of individual runs early? (LogisticRegression has the ability to "warm_start", so we leverage that for incremental training).

**Comment by rohan-gt** [comment]
@richardliaw stop the hyperparameter sweep

**Comment by inventormc** [comment]
In general, we need to be able to look at some metric after each epoch to use Ray Tune's schedulers/early stopping algorithms to stop a hyperparameter sweep early. This is why we currently only early stop on estimators that have `partial_fit` or `warm_start` -- we can look at the metric after each epoch. Other sklearn estimators will just fit all the way to completion without giving us a chance to look at metrics in between epochs.

**Comment by richardliaw** [comment]
Hmm yeah; I think perhaps there is value to stopping the hyperparameter tuning if the top score is converged across the last X trials though (even before having fully evaluated all `n_trials` trials).

**Comment by rohan-gt** [comment]
@richardliaw exactly. You just need to look at the CV score progression

**Comment by rohan-gt** [comment]
In the graph below I'm taking the cumulative max of the CV score as the trials progress. Here we can see that one optimum is reached after 8 trials and we can possibly end the optimization after checking a few trials after that (maybe at 15)

![Screenshot 2020-11-10 at 12 11 15 AM](https://user-images.githubusercontent.com/2353212/98583844-0c108400-22eb-11eb-8d7e-a4c5332ccd9e.png)
